app/views/dams_objects/_admin_download.html.erb

```diff
@@ -1,14 +1,19 @@
 <div id="etc-menu">
 
-	<% ucsd_local_display = (@document['otherRights_tesim'] != nil && @document['otherRights_tesim'].first.to_s.include?('localDisplay')) ? 'true' : 'false' %>
-	<% adl_glyph =  '<i class="glyphicon glyphicon-download-alt"></i>'.html_safe %>
-	<% adl_title = 'Download File' %>
-	<% adl_class = 'adl pull-right' %>
+  <% local_otherRights = (@document['otherRights_tesim'] && @document['otherRights_tesim'].first.to_s.include?('localDisplay')) ? 'true' : 'false' %>
+  <% local_license = 'false'
+     if @document['resource_type_tesim'] && @document['resource_type_tesim'].first.to_s.include?('video')
+       local_license = (@document['license_tesim'] && @document['license_tesim'].first.to_s.include?('localDisplay') ) ? 'true' : 'false'
+     end 
+  %>
+  <% adl_glyph =  '<i class="glyphicon glyphicon-download-alt"></i>'.html_safe %>
+  <% adl_title = 'Download File' %>
+  <% adl_class = 'adl pull-right' %>
 
-	<% if can? :update, @document then %>
-	  <%= link_to adl_glyph, downloadFilePath, :rel => 'nofollow', class:adl_class, title:adl_title %>
-	<% elsif ucsd_local_display == 'false' && !downloadDerivativePath.nil?%>
-	  <%= link_to adl_glyph, downloadDerivativePath, :rel => 'nofollow', class:adl_class, title:adl_title %>
-	<% end %>
+  <% if can? :update, @document then %>
+    <%= link_to adl_glyph, downloadFilePath, :rel => 'nofollow', class:adl_class, title:adl_title %>
+  <% elsif local_otherRights == 'false' && local_license == 'false' && !downloadDerivativePath.nil?%>
```

Comment:
Could we potentially move this code in a helper method, something like `can_download?` (which takes in a solr document) or something? This would include the code above that checks the various Solr properties to set the values of `local_otherRights` and `local_license`

My thinking behind this is because while this code will do a good job preventing links from showing to users who shouldn't see them, I don't think this actually prevents those users from downloading the files directly through the file dowload controller.

So if [this line in the file controller](https://github.com/ucsdlib/damspas/blob/develop/app/controllers/file_controller.rb#L46) could be changed to something like:

`authorize!(:show, @solr_doc) && can_download?(@solr_doc)` 

Then we could enforce the logic in both places with the same code. Or something like that.

Curious what you think.

Reply:
@mcritchlow - Sounds good.  I'll move the code in a helper method and update the file controller.  I think the issue of downloading the files directly is also mentioned in this issue https://github.com/ucsdlib/damspas/issues/467 and Longshou is working on it.

Reply:
@VivianChu - sounds great. Yeah I think this would hopefully take of both tickets in one PR potentially.